### PR TITLE
Implements transpose & add property T for Index/MultiIndex

### PR DESCRIPTION
**Pull request by itholic**
property `T` for Index & MultiIndex
(https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.T.html#pandas.Index.T)

i think `T` for Index/MultiIndex just returns itselves in pandas like below.

```python
>>> idx = pd.Index(['a', 'b', 'c'])
>>> idx
Index(['a', 'b', 'c'], dtype='object')

>>> idx.T
Index(['a', 'b', 'c'], dtype='object')

>>> idx is idx.T
True
```

For MultiIndex

```python
>>> midx = pd.MultiIndex.from_tuples([('a', 'x'), ('b', 'y'), ('c', 'z')])
>>> midx
MultiIndex([('a', 'x'),
            ('b', 'y'),
            ('c', 'z')],
           )

>>> midx.T
MultiIndex([('a', 'x'),
            ('b', 'y'),
            ('c', 'z')],
           )

>>> midx is midx.T
True
```

### CODE REVIEW

File: databricks/koalas/indexes.py [hunk 1]
```diff
@@ -120,6 +120,37 @@ def size(self) -> int:
         """
         return len(self._kdf)  # type: ignore
 
+    @property
+    def T(self):
```
Comment (HyukjinKwon) [inline databricks/koalas/indexes.py]: Can you implement `transpose` and reuse it for `T`? `    T = property(transpose)`.

File: databricks/koalas/missing/indexes.py [hunk 2]
```diff
@@ -30,7 +30,6 @@ def unsupported_property(property_name, deprecated=False, reason=""):
 class _MissingPandasLikeIndex(object):
 
     # Properties
-    T = unsupported_property('T')
```
Comment (HyukjinKwon) [inline databricks/koalas/missing/indexes.py]: Should remove in `_MissingPandasLikeMultiIndex:` too

File: databricks/koalas/indexes.py (hunk 1 above)
Comment (itholic) [reply databricks/koalas/indexes.py]: ah, it would be better. okay i go for it

File: databricks/koalas/missing/indexes.py (hunk 2 above)
Comment (itholic) [reply databricks/koalas/missing/indexes.py]: thanks for the comment!

File: databricks/koalas/indexes.py (hunk 1 above)
Comment (itholic) [reply databricks/koalas/indexes.py]: ah, i just realized that pandas has no `transpose` for `Index`.
then maybe would it be better to keep this behavior??

Comment (HyukjinKwon) [reply databricks/koalas/indexes.py]: I think pandas has it ..

```python
>>> import pandas as pd
>>> pd.DataFrame({'a': [1,2,3]}).index.transpose()
RangeIndex(start=0, stop=3, step=1)
```

Comment (itholic) [reply databricks/koalas/indexes.py]: oops... then maybe they forgot to upload this to their docs.

스크린샷 2019-11-19 오후 3 50 51

thanks i'm going to add it !

File: docs/source/reference/indexing.rst [hunk 3]
```diff
@@ -23,6 +23,8 @@ Properties
    Index.names
    Index.ndim
    Index.empty
+   Index.T
+   Index.transpose
```
Comment (HyukjinKwon) [inline docs/source/reference/indexing.rst]: Ah, then let's exclude `transpose` in the API documentation to be consistent.

Comment (itholic) [reply docs/source/reference/indexing.rst]: okay i just excluded them 👍

**Comment by HyukjinKwon** [comment]
ah, sorry can you resolve conflicts @itholic ?

**Comment by itholic** [comment]
@HyukjinKwon sure, i did 😄

**Pull request merged**